documentation/add/07-sqw_redesign.md

```diff
@@ -0,0 +1,541 @@
+# SQW
+
+## Overview
+
+This section describes the the role of the SQW object within the Horace framework, data it holds and operations it supports without focusing on any implementation details. 
+
+Where specific fields are cited they are the key data that represent that information.
+
+### What is the purpose?
+
+Holds neutron scattering experiment data and provides methods that manipulate, slice and project the data and generate model fits using third-party functions, 
+
+Provide an interface to the Horace file which can be interchanged with external applications (* TBI)
+
+The SQW object exists in two distinct forms: 
+
+- "DND" which contains only processed image data
+- "SQW" which is backed by a full detector pixel array.
+
+In an SQW object where pixel data is included, operations are performed on that data and the current image pixel data recalculated from this. For DND objects operations are performed directly on the image.
+
+Historically these objects both operate as 'first class citizens', but that logical distinction is necessary.
+
+### Operations
+
+#### Generate SQW file
+
+Combine multiple experiment [data files](http://horace.isis.rl.ac.uk/Input_file_formats)  (pairs of legacy `.spe` and `.par` files, or `.nxspe` files from Mantid) into a single SQW object. 
+
+1) `.nxspe` / `.spe` (per-run) data files are converted to intermediate `.sqw`-format data files. Pixel data is ordered by "image pixel" buckets
+
+2) Intermediate files are combined on a image-pixel bucket basis to produce the final `.sqw` file. There is *no aggregation of signal/err data* in this step.
+
+#### Read/Write
+
+- Read the file data, but not pixel data
+
+- Read the pixel data or an m-dimensional cut of the pixel data
+- Read the pixel data corresponding to specific image pixels
+- Write the pixel data for specific image pixels
+- Write all pixel data
+- Support for legacy Horace file format and 'new' NeXus / HDF5 format (TBI)
+- Display data to console (1)
+- Export data to ASCII file (1)
+
+**Notes**
+(1): What data subset of should be displayed at the console or in an text  export file?
+
+#### Data manipulation
+
+- Implement basic arithmetic operations on object data. 
+
+- Cut -- extract a N-dimensional subset of pixels
+
+- Projection -- aggregate pixel data onto an M-dimensional surface (M<N). This should support projections onto planes and spherical shells (TBI) and spiral slices (TBI).
```

Comment:
```suggestion
- Projection -- aggregate pixel data onto an M-dimensional surface (M &ge; 0). This should support projections onto planes and spherical shells (TBI) and spiral slices (TBI).
```
I don't think that we want to restrict to dimensional reducing projections. A reorientation is a projection with `M` &equiv; `N` and 'extra' dimensional projections will have `M` > `N`